Hakawai/Mentions/HKWMentionsPlugin.m

```diff
@@ -863,15 +886,37 @@ - (NSString *)mentionsQuery:(NSString *)text location:(NSUInteger)location {
     NSUInteger mostRecentValidControlCharacterLocation = [self mostRecentValidControlCharacterLocation:text beforeLocation:location];
     if (mostRecentValidControlCharacterLocation != NSNotFound) {
         // Query until end of word in which cursor is present (or until cursor if it is at end of word)
-        NSString *const wordAfterCurrentLocation = [HKWMentionsStartDetectionStateMachine wordAfterLocation:location text:text];
-        NSString *substringUntilEndOfWord = [text substringToIndex:location+wordAfterCurrentLocation.length];
-        // Return the rest of the string after the control char as the query
-        NSString *query = [substringUntilEndOfWord substringFromIndex:mostRecentValidControlCharacterLocation + 1];
-        return query;
+        NSUInteger endOfValidWordAfterLocation = [self endOfValidWordAfterLocation:location text:text];
+        if (endOfValidWordAfterLocation != NSNotFound) {
+            NSString *substringUntilEndOfWord = [text substringToIndex:endOfValidWordAfterLocation];
+            // Return the string, including the control char as the query
+            return [substringUntilEndOfWord substringFromIndex:mostRecentValidControlCharacterLocation];
+        }
     }
     return nil;
 }
 
+/**
+ Search backwards in a string for a character in the control character set
+
+ @param text The text in which to perform a backwards search for a control character
+ @returns Location for most recent control character in a string
+ */
+- (NSUInteger)mostRecentControlCharacterLocation:(NSString *)text {
```

Comment:
can be static utils, which takes text & controlCharacterSet

Reply:
also don't we need to care about cursor position?

Reply:
the string that this is passed does take into account cursor position. and fixed to util